metrics/metrics.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2017, Square, Inc.
+
+// Package metrics is a wrapper of github.com/rcrowley/go-metrics.
+package metrics
+
+import (
+	gometrics "github.com/rcrowley/go-metrics"
+)
+
+type Metrics interface {
+	GetOrRegister(string, interface{})
+}
+
+type metrics struct {
+	registry gometrics.Registry
+}
+
+type Counter interface {
+	Inc(int64)
+	Dec(int64)
+	Count() int64
+	Clear()
+}
+
+type counter struct {
+	counter gometrics.Counter
+}
+
+type Timer interface {
+	Time(func())
+	Percentile(float64) float64
+	Mean() float64
+	Count() int64
+}
+
+type timer struct {
+	timer gometrics.Timer
+}
+
+func NewMetrics() Metrics {
+	return &metrics{
+		registry: gometrics.NewRegistry(),
+	}
+}
+
+// Gets an existing metric or creates and registers a new one.
```

Comment:
I usually look for method docs at the interface definition. Granted, the Go report card will complain unless both places are documented, but imho the interface def is the best place because it's, well, _the_ interface def.

Reply:
ok, will update